### PR TITLE
Introduce ARITY to PrimaryKeyTrait

### PR DESCRIPTION
**Pull request by tyt2y3**
This is one of the last breaking changes I want to ship in 1.0

### CODE REVIEW

**Comment by tyt2y3** [comment]
@billy1624 can you help looking at the test failure?

File: src/executor/insert.rs [hunk 1]
```diff
@@ -116,7 +116,7 @@ where
     {
         // so that self is dropped before entering await
         let mut query = self.query;
-        if db.support_returning() && <A::Entity as EntityTrait>::PrimaryKey::iter().count() > 0 {
```
Comment (tyt2y3) [inline src/executor/insert.rs:119]: I believe all our primary keys has len > 0, so this check is redundant?

Comment (billy1624) [reply src/executor/insert.rs:119]: Redundant indeed!

**Comment by tyt2y3** [comment]
Closed in favour of #2185

**Pull request closed without merging**